clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaConsumerServiceDelegator.java

```diff
@@ -175,16 +174,21 @@ public void batchUnsubscribe(PubSubTopic versionTopic, Set<PubSubTopicPartition>
 
   @Override
   public boolean hasAnySubscriptionFor(PubSubTopic versionTopic) {
-    return consumerServices.stream()
-        .anyMatch(kafkaConsumerService -> kafkaConsumerService.hasAnySubscriptionFor(versionTopic));
+    for (KafkaConsumerService kafkaConsumerService: this.consumerServices) {
```

Comment:
I wonder if you could consider this? This could reduce iterations to just reading atomic integer 
> this could be as simple as tracking the subscription of individual SITs within the class, rather than iterating through all consumer services

Reply:
I would prefer not to. Duplicating state introduces maintenance burden to keep it in sync, and it's not worth it in this case.

Reply:
Sounds reasonable